coffee-jpa/src/main/java/hu/icellmobilsoft/coffee/jpa/health/DatabaseHealth.java

```diff
@@ -0,0 +1,160 @@
+            throw new InvalidParameterException("builderName is mandatory!");
+        }
+        if (StringUtils.isBlank(dataBaseResourceConfig.getDatasourceUrl())) {
+            throw new InvalidParameterException("datasourceUrl is mandatory!");
+        }
+
+        String builderName = dataBaseResourceConfig.getBuilderName();
+        String datasourceUrl = dataBaseResourceConfig.getDatasourceUrl();
+
+        // handle not mandatory configs with defaults
+        String datasourcePrefix = StringUtils.isBlank(dataBaseResourceConfig.getDatasourcePrefix())
+                ? DatabaseHealthConstant.Database.Wildfly.DEFAULT_DATASOURCE_PREFIX
+                : dataBaseResourceConfig.getDatasourcePrefix();
+
+        String dsName = StringUtils.isBlank(dataBaseResourceConfig.getDsName()) ? DatabaseHealthConstant.Database.DEFAULT_DATASOURCE_NAME
+                : dataBaseResourceConfig.getDsName();
+
+        long connectTimeoutSec = dataBaseResourceConfig.getConnectTimeoutSec() == null ? HealthConstant.Common.DEFAULT_CONNECT_TIMEOUT_SEC
+                : dataBaseResourceConfig.getConnectTimeoutSec();
+
+        HealthCheckResponseBuilder builder = HealthCheckResponse.builder().name(builderName);
+        builder.withData(HealthConstant.Common.NODE_NAME, HealthUtil.getNodeId(nodeId));
+
+        String dbUrlValue = config.getValue(datasourceUrl, String.class);
+        builder.withData(HealthConstant.Common.URL, dbUrlValue);
+
+        DataSource datasource = getDataSource(datasourcePrefix, dsName);
+        if (datasource == null) {
+            builder.down();
+            return builder.build();
+        }
+
+        try {
+            connectWithTimeout(datasource, connectTimeoutSec, TimeUnit.SECONDS);
+            builder.up();
+        } catch (InterruptedException e) {
+            builder.down();
+            Thread.currentThread().interrupt();
+            Logger.getLogger(DatabaseHealth.class).error("Error occurred while establishing connection: " + e.getLocalizedMessage(), e);
+        } catch (Exception e) {
+            Logger.getLogger(DatabaseHealth.class).error("Error occurred while establishing connection: " + e.getLocalizedMessage(), e);
+            builder.down();
+        }
+        return builder.build();
+    }
+
+    private DataSource getDataSource(String datasourcePrefix, String dsName) {
+        try {
+            Context initContext = new InitialContext();
+            return (DataSource) initContext.lookup(datasourcePrefix + dsName);
+        } catch (NamingException e) {
+            Logger.getLogger(DatabaseHealth.class).error("Error occured while getting datasource: " + e.getLocalizedMessage(), e);
+        }
+        return null;
+    }
+
+    private Connection connectWithTimeout(DataSource datasource, long timeout, TimeUnit unit)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
Ezt az executor-t (és az 1 szálat) ki zárja le?

Reply:
Amennyire láttam ez ebben a formában a garbage collector miatt automatan lezarodik, és magában meg is szünteti a létrehozott szálat, kb 10 thread ameddig fel megy terhelés alatt is, ez azért is van mert az app server is csak 2 szalat ad defaultbol ami a management porton a health szamara elerhető (nem ugy mint a restnel akar több mint 100 is lehet). Mindenesetre átírtam hogy ne hozzon létre mindig executort a kód, így még biztonságosabb.